src/pobox.erl

```diff
@@ -491,7 +496,7 @@ drop(stack, N, Size, L) ->
        Size =< N -> []
     end;
 drop(keep_old, N, Size, Queue) ->
-    if Size > N  -> element(1, queue:split(N, Queue));
+    if Size > N  -> element(1, queue:split(Size - N, Queue));
```

Comment:
This fixes the resize issue. I guess this explains why it was happening only in `keep_old`.

Reply:
Nice. That was a tricky rare-to-trigger bug.